### DISCUSSION TITLE
Ability to define selector for SearchIndexingSubscriber

### DISCUSSION THREAD

**Discussion by jono-allen** [medusajs/medusa#4735] [Feature Requests]
Currently, the SearchIndexingSubscriber will return all products regardless of status. 

This is not desired in all cases. It does not allow to filter out products that aren't published as an example.

We would like to propose adding a configuration in the search plugins, like an Algolia, to allow selectors to be passed into the retrieveNextProducts function from the medusa plugins config.

The proposed change would look something like 

```
// packages/medusa/src/subscribers/search-indexing.ts
return await this.productService_.list(
       ...this.searchService_.options?.selectors // new
      { id: { gt: lastSeenId },
       },
      {
        relations,
        take: take,
        order: { id: "ASC" },
      }
    )
```
```
// medusa-plugin-algolia/src/types.ts
export type AlgoliaPluginOptions = {
  applicationId: string
  adminApiKey: string
  /**
   * Index settings
   */
  settings?: {
    [key: string]: SearchTypes.IndexSettings
  }
  selector?: ProductSelector // new
}

```
Would love to get feedback and then proceed to open a pr.

Thanks

**Comment by tsvetann** [comment]
Hey what do orders have to do with search subscribers? On another note you can use the transformer function to filter out products as you wish. Is the transformer function not enough to achieve your goal?
